source/Calamari/Kubernetes/ManifestReporter.cs

```diff
@@ -24,20 +24,25 @@ public class ManifestReporter : IManifestReporter
         readonly IVariables variables;
         readonly ICalamariFileSystem fileSystem;
         readonly ILog log;
+        readonly IApiResourceScopeLookup resourceScopeLookup;
 
         static readonly ISerializer YamlSerializer = new SerializerBuilder()
-                                                     .Build();
+            .Build();
 
-        public ManifestReporter(IVariables variables, ICalamariFileSystem fileSystem, ILog log)
+        public ManifestReporter(IVariables variables, ICalamariFileSystem fileSystem, ILog log, IApiResourceScopeLookup resourceScopeLookup)
         {
             this.variables = variables;
             this.fileSystem = fileSystem;
             this.log = log;
+            this.resourceScopeLookup = resourceScopeLookup;
         }
 
         string GetNamespace(YamlMappingNode yamlRoot)
         {
-            var implicitNamespace = variables.Get(SpecialVariables.Namespace) ?? "default";
+            //we check to see if there is an explicit helm namespace defined first
+            //then fallback on the action/target default namespace
+            //otherwise fallback on default
+            var implicitNamespace = variables.Get(SpecialVariables.Helm.Namespace) ?? variables.Get(SpecialVariables.Namespace) ?? "default";
```

Comment:
Helm steps can have their own default namespace, so fallback on that first, then the target/action namespace, then the cluster default